spec/core/hard_machine_spec.rb

```diff
@@ -0,0 +1,83 @@
+require "rspec"
+require "core/game"
+require "core/board"
+require "core/hard_machine"
+
+RSpec.describe HardMachine do
+  it "chooses a winning move when it has the occasion" do
+    move = get_best_move([
+      "X", " ", "X",
+      " ", "O", " ",
+      " ", " ", "O"
+    ])
+
+    expect(move).to eql(1)
+  end
+
+  it "chooses a winning move when it has the occasion (II)" do
+    move = get_best_move([
+      "X", " ", " ",
+      " ", "O", " ",
+      "X", " ", "O"
+    ])
+
+    expect(move).to eql(3)
+  end
+
+  it "chooses a blocking move when the opponent is going to win in the following turn" do
+    move = get_best_move([
+      " ", " ", "X",
+      " ", "O", " ",
+      "X", " ", "O"
+    ])
+
+    expect(move).to eql(0)
+  end
+
+  it "chooses a blocking move when the opponent is going to win in the following turn (II)" do
+    move = get_best_move([
+      " ", " ", "X",
+      "X", " ", "O",
+      "O", " ", "O"
+    ])
+
+    expect(move).to eql(7)
+  end
+
+  it "draws a game against another HardMachine" do
+    player1 = HardMachine.new("X")
+    player2 = HardMachine.new("O")
+    game = Game.new(player1, player2)
+
+    play_game(game)
+
+    expect(game.ended?).to eql(true)
+    expect(game.winner).to eql(nil)
+  end
+
+  it "wins a game against another HardMachine which has made a mistake" do
+    player1 = HardMachine.new("X")
+    player2 = HardMachine.new("O")
+    game = Game.new(player1, player2)
+    game.board = Board.create_from_array([
+      "X", " ", " ",
+      "O", " ", " ",
+      " ", " ", " "
+    ])
+
+    play_game(game)
+
+    expect(game.ended?).to eql(true)
```

Comment:
rspec lets you use .to which enables you to make this read much better, something like expect(game).to be ended, which will also help you name your method correct. I'd change ended? to over? so that it reads better.